pmd-java/src/main/java/net/sourceforge/pmd/lang/java/ast/AbstractJavaAccessNode.java

```diff
@@ -135,4 +142,79 @@ private void setModifier(boolean enable, int mask) {
     public boolean isPackagePrivate() {
         return !isPrivate() && !isPublic() && !isProtected();
     }
+
+    /**
+     * Get all annotations present on this node.
+     *
+     * @return all annotations present on this node.
+     */
+    @Override
+    public List<ASTAnnotation> getDeclaredAnnotations() {
+        List<ASTAnnotation> result = new ArrayList<>();
+        Node parent = this.jjtGetParent();
+        for (int index = 0; index < parent.jjtGetNumChildren(); index++) {
+            Node child = parent.jjtGetChild(index);
+            if (child == this) {
+                return result;
+            } else if (child instanceof ASTAnnotation) {
+                result.add((ASTAnnotation) child);
+            }
+        }
+        return result;
+    }
+
+    /**
+     * Get specific annotaion on this node.
+     *
+     * @param annotQualifiedName
+     *            qulified name of the annotation.
+     * @return <code>ASTAnnotaion</code> node if the annotation is present on this node, else <code>null</code>
+     */
+    @Override
+    public ASTAnnotation getSpecificAnnotation(String annotQualifiedName) {
```

Comment:
Maybe name that just `getAnnotation` ? The javadoc description has some typos too